src/service.py

```diff
@@ -14,166 +14,18 @@
-            self._set_response(200, str(trial_number))
-        else:
-            self._set_response(400, "-1")
-
-    def handle_generate_subsequent_operation(self, json_object):
-        """Process EXP_TRIAL_GENERATE_SUBSEQUENT operation."""
-        is_valid_json_object = validate_trial_generate_json(json_object)
-        experiment_id = json_object["experiment_id"]
-        if is_valid_json_object and hpo_service.instance.containsExperiment(experiment_id):
-            trial_number = hpo_service.instance.get_trial_number(experiment_id)
-            self._set_response(200, str(trial_number))
-        else:
-            self._set_response(400, "-1")
-
-    def handle_result_operation(self, json_object):
-        """Process EXP_TRIAL_RESULT operation."""
-        if (hpo_service.instance.containsExperiment(json_object["experiment_id"]) and
-                json_object["trial_number"] == hpo_service.instance.get_trial_number(json_object["experiment_id"])):
-            hpo_service.instance.set_result(json_object["experiment_id"], json_object["trial_result"], json_object["result_value_type"],
-                       json_object["result_value"])
-            self._set_response(200, "0")
-        else:
-            self._set_response(400, "-1")
-
-
-def get_search_create_study(search_space_json, operation):
-    # TODO: validate structure of search_space_json
-    
-    if operation == "EXP_TRIAL_GENERATE_NEW":
-        if "parallel_trials" not in search_space_json:
-            search_space_json["parallel_trials"] = n_jobs
-        experiment_name, total_trials, parallel_trials, direction, hpo_algo_impl, id_, objective_function, tunables, value_type = get_all_tunables(
-            search_space_json)
-        if (not parallel_trials):
-            parallel_trials = n_jobs
-        elif parallel_trials != 1:
-            raise Exception("Parallel Trials value should be '1' only!")
-
-        logger.info("Total Trials = "+str(total_trials))
-        logger.info("Parallel Trials = "+str(parallel_trials))
-        
-        if hpo_algo_impl in ("optuna_tpe", "optuna_tpe_multivariate", "optuna_skopt"):
-            hpo_service.instance.newExperiment(id_, experiment_name, total_trials, parallel_trials, direction, hpo_algo_impl, objective_function,
-                                                 tunables, value_type)
-            print("Starting Experiment: " + experiment_name)
-            hpo_service.instance.startExperiment(id_)
-
-
-def get_search_space(id_, url):
-    """Perform a GET request and return the search space json."""
-    params = {"id": id_}
-    r = requests.get(url, params)
-    r.raise_for_status()
-    search_space_json = r.json()
-    return search_space_json
-
-
-
-def main():    
-    server = HTTPServer((host_name, server_port), HTTPRequestHandler)
-    logger.info("Access server at http://%s:%s" % ("localhost", server_port))
-    server.serve_forever()
+    restService.join()
+    gRPCservice.join()
```

Comment:
We will need to update all the module dependencies for gRPC in requirements.txt & in the docker file

Reply:
This has been resolved in Dockerfile and requirements.txt